S4TF_Tutorial_5.ipynb

```diff
@@ -0,0 +1,1181 @@
+        "print(\"total training dog images: \\(dogTrainList.count)\")\n",
+        "\n",
+        "print(\"total validation cat images: \\(catTestList.count)\")\n",
+        "print(\"total validation dog images: \\(dogTestList.count)\")\n",
+        "print(\"--\")\n",
+        "print(\"Total training images: \\(trainList.count)\")\n",
+        "print(\"Total validation images: \\(testList.count)\")\n",
+        "\n",
+        "print(Python.type(trainList))"
+      ],
+      "execution_count": 8,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "total training cat images: 1000\r\n",
+            "total training dog images: 1000\r\n",
+            "total validation cat images: 500\n",
+            "total validation dog images: 500\n",
+            "--\n",
+            "Total training images: 2000\n",
+            "Total validation images: 1000\n",
+            "<class 'list'>\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "akDk2rmO70un",
+        "colab_type": "text"
+      },
+      "source": [
+        "# Data Preparation "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nQ7KrBqH71L0",
+        "colab_type": "text"
+      },
+      "source": [
+        "Images must be formatted into appropriately pre-processed floating point tensors before being fed into the network. The steps involved in preparing these images are:\n",
+        "\n",
+        "1. Read images from the disk\n",
+        "2. Decode contents of these images and convert it into proper grid format as per their RGB content\n",
+        "3. Convert them into floating point tensors\n",
+        "4. Rescale the tensors from values between 0 and 255 to values between 0 and 1, as neural networks prefer to deal with small input values.\n",
+        "\n",
+        "We have done this in the following code."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "aRAT9M538M82",
+        "colab_type": "text"
+      },
+      "source": [
+        "The `getTensor` function takes image path as input and outputs a tuple of resized image tensor and it’s corresponding label.\n",
+        "\n",
+        "The `getTensorArray` function takes in a list of paths and the number of tensors to be produced in the output tensor as input. While processing, it gives number of images done as print statements so that we can have an understanding of how much process is done."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "1DC85ego6kmK",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "// getTensor returns the resized image tensor and it's corresponding label\n",
+        "\n",
+        "func getTensor(path: String) -> (Tensor<Float>, Int32) {\n",
```

Comment:
The word "get" (or "set") is not quite idiomatic in Swift names. Swift functions should be named according to their side effects. Functions that are not side-effecting should read like noun phrases.

For example, this function should be named `tensor(fromPath:)`. Same for all other occurrences of "get".

Source: [Swift API Design Guidelines - Strive for Fluent Usage](https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage).